Fix 64bit indices build

Pull request #4778 broke 64bit indices configurations:

```
source/hp/dof_handler.cc:1650:24: error: no match for ‘operator=’ (operand types are ‘std::vector<unsigned int>’ and ‘std::remove_reference<std::vector<long long unsigned int> >::type {aka std::vector<long long unsigned int>}’)

     vertex_dof_offsets = std::move (std::vector<types::global_dof_index>());
```

[1] https://cdash.kyomu.43-1.org/viewBuildError.php?buildid=9995

In reference to #4704

--- a/source/hp/dof_handler.cc
+++ b/source/hp/dof_handler.cc
@@ -1647,7 +1647,7 @@ namespace hp
     faces.reset ();
 
     vertex_dofs = std::move(std::vector<types::global_dof_index>());
-    vertex_dof_offsets = std::move (std::vector<types::global_dof_index>());
+    vertex_dof_offsets = std::move (std::vector<unsigned int>());
   }
 }
 

